force-app/main/default/classes/ERR_Notifier.cls

```diff
@@ -28,162 +28,255 @@
+     * @description Just in case the size of the Error__c records (LOB fields) plus the size of
+     * the created Email Message Body or Chatter Post Body is within 95% of the total heap available
+     * stop building the message/post body and send/post with the content as is. This avoids an
+     * untrappable limit exception.
+     */
+    private Boolean isNearHeapLimit() {
+        return (Limits.getHeapSize() >= MAX_HEAP_LIMIT);
+    }
+
     /*******************************************************************************************************
-    * @description Inner class with the 3 types of notification options: user, chatter group, and all system
-    *              administrators.
-    */
+     * @description Inner class with the 3 types of notification options: user, chatter group, and all system
+     *              administrators.
+     */
     public class NotificationOptions {
         public final String sysAdmins = 'All Sys Admins';
-        public final String user = '005';  //specific user (should be a system admin)
+        public final String user = '005'; //specific user (should be a system admin)
         public final String chatterGroup = '0F9'; //chatter group
     }
 
-    /*******************************************************************************************************
-    * @description Sends error notifications to the receivers specified in the settings, if error notifications are enabled, with
-    *              all the existing error records that have not been included in previous error notifications.
-    * @param context The context that triggered the error notification.
-    * @return void
-    */
-    public static void sendErrorNotifications(String context) {
+    /**********************************************************************************************
+     * @description Using the provided error notification user id (or 'All System Admins') or gets
+     * all system admin email addresses.
+     * @param errorNotifRecipient Id or Null
+     * @return List<Email>
+     */
+    @TestVisible
+    private List<String> getErrorEmailRecipients(String errorNotifRecipient) {
+        List<String> sendList = new List<String>();
+        if (errorNotifRecipient instanceof id && errorNotifRecipient.startsWith(NotificationOptions.user)) {
+            List<User> useremaillist = new List<User>();
+            useremaillist = [SELECT email FROM User WHERE id = :errorNotifRecipient AND isActive = TRUE];
+            for (User u : useremaillist) {
+                sendList.add(u.email);
+            }
+        } else if (errorNotifRecipient == NotificationOptions.sysAdmins) {
+            sendList.addAll(this.getSystemAdminEmails());
+        }
+        return sendList;
+    }
 
+    /**********************************************************************************************
+     * @description Get the active System Administrators' emails.
+     * @return List<String> Emails
+     */
+    @TestVisible
```

Comment:
I didn't see this referenced in the test class, is this annotation needed?

Reply:
Right, this one is not needed, removed.